docs/dev-process/coding-guidelines.md

```diff
@@ -1,92 +1,179 @@
-
-# C# Coding Style
+# C# Coding Guidelines
 
 ## Coding Conventions
 
 As a general rule, our coding convention is to follow the style of the surrounding code.
-Avoid reformatting any code when submitting a PR as it obscures the functional changes of your change.
-We run the [.NET code formatter tool](https://github.com/dotnet/codeformatter) regularly help keep consistent formatting.
+So if a file happens to differ in style from conventions defined here
+(e.g. private members are named `m_member` rather than `_member`),
+the existing style in that file takes precedence.
+
+When making changes, you may find some existing code goes against the conventions defined here.
+In such cases, please avoid reformatting any existing code when submitting a PR as it obscures the functional changes of the PR.
+A separate PR should be submitted for style-only changes.
```

Comment:
Here are my experiences:

1. Depending on how bad the reformatting affects the review, I will choose to either tolerate the formatting changes or block the PR and ask the author to revert the formatting changes.
    * For example, some PRs has changes that remove the trailing spaces. When that kind of formatting changes doesn't affect review much, I will tolerate them.
    * But for some other PRs, the formatting changes might cause the review to be difficult, and I will block the PR and ask the author to revert formatting changes.

2. We ask authors to revert the formatting changes and suggest them to submit a separate PR, but they are not required to submit a PR to fix the formatting. You can give them the choices as for how to revert the reformatting changes -- submit style PR and merge and then rebase the original PR is certainly one way to do it; or they can just revert the formatting changes and submit a style PR later.

3. I don't like tracking issue in this situation.

Reply:
I unlike tracking issue too. But without it we can forget and lost. So we should open style PR before continue original PR.

Reply:
I don't think we should make it a requirement for the author to submit a style-only PR in this situation. It's acceptable if they just revert the formatting changes, and it would be a plus if they are willing to go one step further to fix the style/formatting as well.